File: tests/unit/server/config.test.ts
import { describe, expect, it } from 'vitest';
import { resolveConfig } from '../../../server/src/config.js';

describe('resolveConfig', () => {
  it('binds localhost with a fresh random token by default', () => {
    const a = resolveConfig();
    const b = resolveConfig();
    expect(a.host).toBe('127.0.0.1');
    expect(a.port).toBe(3002);
    expect(a.token).toHaveLength(32);
    expect(a.token).not.toBe(b.token);
  });

  it('uses the dev token when provided', () => {
    expect(resolveConfig({ devToken: 'dev' }).token).toBe('dev');
  });

  it('applies overrides', () => {
    const config = resolveConfig({ port: 0, openBrowser: false, staticRoot: '/tmp/x' });
    expect(config.port).toBe(0);
    expect(config.openBrowser).toBe(false);
    expect(config.staticRoot).toBe('/tmp/x');
  });
});

File: app/internal/server/config_test.go
package server

import "testing"

// Cases ported from tests/unit/server/config.test.ts.
func TestResolveConfigDefaults(t *testing.T) {
	a := ResolveConfig(Overrides{})
	b := ResolveConfig(Overrides{})
	if a.Host != "127.0.0.1" {
		t.Fatalf("host = %q", a.Host)
	}
	if a.Port != 3002 {
		t.Fatalf("port = %d", a.Port)
	}
	if len(a.Token) != 32 {
		t.Fatalf("token length = %d, want 32", len(a.Token))
	}
	if a.Token == b.Token {
		t.Fatal("tokens must be fresh per run")
	}
}

func TestResolveConfigDevToken(t *testing.T) {
	if got := ResolveConfig(Overrides{DevToken: "dev"}).Token; got != "dev" {
		t.Fatalf("token = %q, want dev", got)
	}
}

func TestResolveConfigOverrides(t *testing.T) {
	port := 0
	openBrowser := false
	cfg := ResolveConfig(Overrides{Port: &port, OpenBrowser: &openBrowser, StaticRoot: "/tmp/x"})
	if cfg.Port != 0 {
		t.Fatalf("port = %d, want 0", cfg.Port)
	}
	if cfg.OpenBrowser {
		t.Fatal("openBrowser must be overridable to false")
	}
	if cfg.StaticRoot != "/tmp/x" {
		t.Fatalf("staticRoot = %q", cfg.StaticRoot)
	}
}
